src/main/java/domain/GameResult.java

```diff
@@ -0,0 +1,39 @@
+package domain;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map.Entry;
+
+public class GameResult {
+
+    private final HashMap<Member, Reward> gameResult;
+
+    public GameResult() {
+        this.gameResult = new LinkedHashMap<>();
+    }
+
+    public void addGameResult(Member member, Reward reward) {
+        gameResult.put(member, reward);
+    }
+
+    public GameResultDto getResultByTarget(ResultTarget target) {
+        if (target.isAllMembers()) {
+            return getResultOfAllMembers();
+        }
+        return getResultOfMember(target.getValue());
+    }
+
+    private GameResultDto getResultOfAllMembers() {
+        return new GameResultDto(gameResult);
+    }
+
+    private GameResultDto getResultOfMember(String memberName) {
+        HashMap<Member, Reward> resolvedResult = new LinkedHashMap<>();
+        Entry<Member, Reward> memberResult = gameResult.entrySet().stream()
+                .filter(entry -> entry.getKey().hasSameNameWith(memberName))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("해당 이름을 가진 참여자가 없습니다."));
+        resolvedResult.put(memberResult.getKey(), memberResult.getValue());
+        return new GameResultDto(resolvedResult);
+    }
```

Comment:
```suggestion
    private GameResultDto getResultOfMember(String memberName) {
        HashMap<Member, Reward> resolvedResult = new LinkedHashMap<>();
        Member member = new Member(memberName);
        if (!gameResult.containsKey(member)) {
            throw new IllegalArgumentException("해당 이름을 가진 참여자가 없습니다.");
        }
        Reward reward = gameResult.get(member);

        resolvedResult.put(member, reward);
        return new GameResultDto(resolvedResult);
    }
```

이렇게 쓰면 어떨까요? HashMap이라는 자료형을 사용하는데, 굳이 루프를 돌면서 찾을 필요가 없을 것 같아요!
HashMap에서 요소를 찾는 시간복잡도는 O(1)이지만 루프를 시도하는 경우 O(N)이 되어버리네요!
가독성 측면에서도 더 나아보이는데 에버는 어떻게 생각하시나요?!🤔

Reply:
그렇네요..
매번 리스트를 사용하다가 맵을 사용해서
당연하게 루프를 돌린 것 같아요 😂
좋은 지적 감사합니다!